LocationManager/INTULocationManager/INTULocationManager.m

```diff
@@ -709,7 +709,7 @@ - (void)locationManager:(CLLocationManager *)manager didFailWithError:(NSError *
     INTULMLog(@"Location services error: %@", [error localizedDescription]);
     self.updateFailed = YES;
     
-    [self completeAllLocationRequests];
+    [self processLocationRequests];
```

Comment:
Gotcha, thanks for the detail. Would something like this suffice for the behavior you're describing:

```
- (void)locationManager:(CLLocationManager *)manager didFailWithError:(NSError *)error
{
    INTULMLog(@"Location services error: %@", [error localizedDescription]);
    self.updateFailed = YES;

    for (INTULocationRequest *locationRequest in self.locationRequests) {
        if (locationRequest.isRecurring) {
            // Keep the recurring request alive
            [self processRecurringRequest:locationRequest];
        } else {
            // Fail any non-recurring requests
            [self completeLocationRequest:locationRequest];
        }
    }
}
```